server/datastore/mysql/software_titles.go

```diff
@@ -217,10 +218,21 @@ func (ds *Datastore) ListSoftwareTitles(
 
 	for _, p := range policies {
 		if i, ok := titleIndex[p.TitleID]; ok {
-			if softwareList[i].AppStoreApp != nil {
-				softwareList[i].AppStoreApp.AutomaticInstallPolicies = append(softwareList[i].AppStoreApp.AutomaticInstallPolicies, p)
-			} else {
-				softwareList[i].SoftwarePackage.AutomaticInstallPolicies = append(softwareList[i].SoftwarePackage.AutomaticInstallPolicies, p)
+			switch {
+			case softwareList[i].AppStoreApp != nil:
+				softwareList[i].AppStoreApp.AutomaticInstallPolicies = append(
+					softwareList[i].AppStoreApp.AutomaticInstallPolicies, p,
+				)
+			case softwareList[i].SoftwarePackage != nil:
```

Comment:
Makes sense to be defensive and add the check here.

Curious if any of the other places suffer from this issue. I just recently added this into `ListHostSoftware` https://github.com/fleetdm/fleet/blob/main/server/datastore/mysql/software.go#L4294-L4300

Think it's worth updating there too?

Reply:
Added there too just in case.